modules/raft/src/test/java/org/apache/ignite/internal/raft/client/PhysicalTopologyAwareRaftGroupServiceRunTest.java

```diff
@@ -436,6 +472,217 @@ void testBoundedTimeoutSucceedsWhenLeaderAppearsBeforeTimeout() {
         assertThat(result, willCompleteSuccessfully());
     }
 
+    /**
+     * Tests that with bounded timeout, the command fails if peer responses are slow and exceed the deadline.
+     */
+    @Test
+    void testBoundedTimeoutExpiresWhileTryingPeers() {
+        // Each peer response takes 200ms and returns EPERM (no leader).
+        // With 3 peers and 500ms timeout, we should exceed the deadline.
+        when(messagingService.invoke(
+                any(InternalClusterNode.class),
+                argThat(this::isTestWriteCommand),
+                anyLong()
+        )).thenAnswer(invocation -> {
+            return CompletableFuture.supplyAsync(() -> {
+                try {
+                    Thread.sleep(200);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+                return FACTORY.errorResponse()
+                        .errorCode(RaftError.EPERM.getNumber())
+                        .build();
+            }, executor);
+        });
+
+        PhysicalTopologyAwareRaftGroupService svc = startService();
+
+        // With 500ms timeout and 3 peers each taking 200ms, total time ~600ms > 500ms.
+        CompletableFuture<Object> result = svc.run(testWriteCommand(), 500);
+
+        assertThat(result, willThrow(ReplicationGroupUnavailableException.class, 2, TimeUnit.SECONDS));
+    }
+
+    /**
+     * Tests that individual peer request timeouts are bounded by the deadline.
+     *
+     * <p>If the deadline is 500ms but the default response timeout is 3000ms, the first peer request
+     * should timeout at ~500ms, not wait the full 3000ms.
+     */
+    @Test
+    void testBoundedTimeoutLimitsIndividualPeerRequestTimeout() {
+        // First peer never responds (simulating network issue).
+        // Default response timeout is 3000ms, but our deadline is 500ms.
+        when(messagingService.invoke(
+                any(InternalClusterNode.class),
+                argThat(this::isTestWriteCommand),
+                anyLong()
+        )).thenReturn(new CompletableFuture<>()); // Never completes
+
+        PhysicalTopologyAwareRaftGroupService svc = startService();
+
+        long startTime = System.currentTimeMillis();
+
+        // With 500ms timeout, should fail within ~600ms (500ms + some margin), not 3000ms.
+        CompletableFuture<Object> result = svc.run(testWriteCommand(), 500);
+
+        assertThat(result, willThrow(ReplicationGroupUnavailableException.class, 1, TimeUnit.SECONDS));
+
+        long elapsed = System.currentTimeMillis() - startTime;
+        // Should complete much faster than the default response timeout (3000ms).
+        assertTrue(elapsed < 1500, "Expected to fail within 1500ms but took " + elapsed + "ms");
+    }
+
+    // ==================== Shutdown Tests ====================
```

Comment:
Er, let's remove this line, test names already contain `Shutdown`

Reply:
done